homeassistant/components/homekit/accessories.py

```diff
@@ -137,6 +138,35 @@ def update_state(self, new_state):
         """
         raise NotImplementedError()
 
+    def sent_logbook_message(self, homekit_event, logbook_message):
+        """Sent event to logbook."""
+        if logbook_message:
+            logbook_attr_message = "{} for {}".format(
+                logbook_message, self.display_name)
+        else:
+            logbook_attr_message = "for {}".format(
+                self.display_name)
+
+        self.hass.bus.fire(homekit_event, {
+            ATTR_ENTITY_ID: self.entity_id,
+            ATTR_MESSAGE: logbook_attr_message
+        })
+
+    def call_service(self, acc_domain, service, params, logbook_message=None):
+        """Sent event for logbook that a change was initiated from homekit.
+
+        Then call the service to change the state.
+        This method is to be called from methods that change states using
+        self.call_service(DOMAIN, service, params, logbook_message).
+        """
+        if logbook_message is not None:
+            message = "send command {}".format(logbook_message)
+        else:
+            message = "send command {}".format(service)
+
+        self.sent_logbook_message(EVENT_HOMEKIT_CHANGED, message)
```

Comment:
Can you combine these two methods into `call_service`? I see no reason to have them sparate.

Reply:
I had them separate to allow other items to send a log message as well (i.e. adding an accessory).
Based on our conversations I believe that we will only create the logbook entry for state changes and anything else would be done by triggering an event and then the user would use automation rules to create the logbook entry. For example adding an accessory.
Combined it now and changed some other parts on it. Have a look. :-)

Reply:
On the logger debug statement. I did not move it to call_service right now. Currently the debug statement is the 1st statement being done, before anything else will be done. Hence if there is any kind of issue or so when going through the method the bug statement would have been written that it entered that method.
If we move it however to call_service then the debug statement will not be written till almost the end. Thus if there is a failure due to an unexpected value the debug statement would  not be written out then.